cmd/auth-rpc-client.go

```diff
@@ -80,13 +80,13 @@ func newAuthRPCClient(config authConfig) *AuthRPCClient {
 
 // Login - a jwt based authentication is performed with rpc server.
 func (authClient *AuthRPCClient) Login() (err error) {
-	authClient.Lock()
-	defer authClient.Unlock()
-
+	authClient.RLock()
 	// Return if already logged in.
 	if authClient.authToken != "" {
+		authClient.RUnlock()
 		return nil
 	}
+	authClient.RUnlock()
```

Comment:
This makes login to be called in parallel by multiple goroutines.

What is the purpose of calling `authClient.rpcClient.Call(serviceMethod, &args, &reply);` in parallel?

In the profiling result of referred issue, I don't see a hit of this method.  As `Login()` is called only once upon connect/reconnect, why it is making performance penalty?